Marlin/src/feature/mmu/mmu2.cpp

```diff
@@ -884,7 +884,11 @@ void MMU2::filament_runout() {
       // Slowly spin the extruder during C0
       else {
         while (planner.movesplanned() < 3)
-          unscaled_mmu2_e_move(0.25, MMM_TO_MMS(120), false);
+          // do nothing here as the previous implementation of
+          // "keep slowly spinning during C0" causes a huge amount of
+          // filament to be extruded and causing catastrophic/unrecoverable
+          // issues...
+          idle();
```

Comment:
Removing the move would seem to negate the point of checking that the planner has fewer than 3 moves. The comment above says this block will "Slowly spin the extruder during C0." Will that still occur with these changes? If not, that comment should also be removed.

The code being changed here was originally added in #19429 by @isanval so it would be a good idea to review that PR and to get their input to understand what this was intended to accomplish and to come to a consensus about the best solution.

Reply:
I read through the afformentioned PR and issue tickets. First of all, the claim of MMU not having enough torque to push the filament into gears doesn't feel right to me, I suspect @isanval had voltage issues mentioned and already solved here: [I found serious design error in MMU2 electronics hardware](https://forum.prusa3d.com/forum/original-prusa-i3-mmu2s-mmu2-hardware-firmware-and-software-help/i-found-serious-design-error-in-mmu2-electronics-hardware/paged/13/#post-689025)

Secondly, in my setup the filament sensor is around 8-10mm before the gears and keep rotating the gears doesn't help for anything. I wish I had a video showing the error.

Maybe I should introduce a new config option like `MMU2_EXTRUDE_WHILE_C0` which True by default and keep both the codes. This way I can fix my problem and keep the previous implementation, what do you think?

Reply:
PR claim that move should be done with MMU2S but if you have filament sensor before gears you should not have MMU2S but MMU2 (AFAIK)

Reply:
> but if you have filament sensor before gears you should not have MMU2S but MMU2 (AFAIK)

That's not true, MMU2 doesn't have a sensor at all and only extrudes a pre-defined length of filament. With my setup MMU2s feeds until the filament sensor triggers, but pushes the filament a little further after that, putting the filament to the gears.

But, sure enough, even if I find a proper way of putting the filament sensor inside my Sprite Extruder Pro (I have an idea how to do that) the gears rotating indefinitely will cause the same issue that I'm currently having...

Reply:
That's true, it's not MMU2 that has such sensor but I recall, maybe wrongly, that together with MMU2S the sensor on extruder was moved "on gears" to be sure filament was got. Maybe just wrong assumption on my side

Reply:
The MMU2S uses a “chimney” filament sensor on the extruder body to verify proper filament swaps: https://help.prusa3d.com/guide/3a-mk3s-extruder-upgrade_188808

Reply:
> with MMU2S the sensor on extruder was moved "on gears" to be sure filament was got

aah I just realized, the main intention of this feature... how stupid I feel right now :) @thinkyhead tried to warn me about this, but I was so focused on my problem... sorry about that!

if we disable this (as I was trying to), the MMU2s will not be able to push the filament behind the gears if the gears are not rotating 🤦 

I think we can either close this PR or as I suggested we can add another configuration option to make this optional with the default behaviour matching the current one...

Reply:
Wouldn’t you just use the `PRUSA_MMU2` option instead of `PRUSA_MMU2S`?

“Requires MK3S extruder with motion sensor”

https://github.com/MarlinFirmware/Marlin/blob/68ab7f6bb7d93ed24054b033334c5821ea4e4cbf/Marlin/Configuration.h#L376-L377

Reply:
Unfortunately no, I still have the benefits of having a filament sensor, as sometimes the pulley cannot pull the filaments for some reason (friction in the spool holders etc.), and MMU extrudes shorter than required and thanks to the filament sensor MMU knows that and starts the intermitent extrusion sequence...

Reply:
If you’re using an MMU2S without the required MK3S extruder (or something closely resembling it), then a hidden flag to disable some MMU2S functionality might be best.

Thinkyhead will need to weigh in.

Reply:
You require the extruder end. not just for the sensor. It requires a specially sized PTFE  that has a ID of 1.85 mm in the entrance of the hotend.. On retraction the hot filament is pulled into this section and allowed to form a blob of maximum diameter 1.85 before being pulled through the rest of the unit.
Using PFTE hotend tubes of other diameters or lengths will cause jams or just fail to work.   It is all very carefully timed for the expected dimensions and lengths

Reply:
Although, I feel that the rotation of the extruder gears while `C0` will still cause issues, and I'll disable that feature on my own branch, I'm convinced that what I'm doing is out of the booundaries of the intended design of MMU2s. So, I think we can reject this PR...

@ellensp I don't have any problems with the tip shapes (thickness or stringiness), I adjusted my slicer for the cooling tube length and position, hopefully the filament tips are coming out perfectly, it took a while to achieve that though...